### DISCUSSION TITLE
.Net core 6 modules running in Oqtane 2.2 (.Net 5)?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#1649] [Q&A] [answered]
If we wanted to run a version ahead with our modules, do you think a .Net 6 module would work properly in the current release version of Oqtane that is .Net 5?

**Comment by sbwalker** [comment]
I highly doubt it. In general, platforms support backward compatibility but not forward compatibility.

**Reply by markdav-is** [reply]
Thanks for the quick reply. We'll wait for sanity's sake.

**Comment by sbwalker** [comment, marked as answer]
The goal is to release Oqtane 3.0 for .NET 6 at the beginning of November. There will be one more release for Oqtane 2.x which will contain all of the latest bits in the Dev branch. Moving to .NET 6 is important as it is an LTS release whereas .NET 5 is a GA release.

**Reply by markdav-is** [reply]
I was guessing that was the case. We have some active projects that we plan to upgrade when the time is right

**Comment by curlyfro** [comment]
i'm waiting for the Identity Server and 3rd party auth.  is that release still going to happen before 3.0?

**Reply by sbwalker** [reply]
I think you are referring to "Authentication extensibility ( ie. Azure B2C, Social logins, etc... )" which is currently identified on the roadmap for the 2.3 release? This item is a fairly heavy lift and no longer appears to be possible within the time frame ( .NET 6 ships in early November ). Please note that the intended scope for this feature was only to allow Oqtane to authenticate against third party identity providers such as Azure B2C, Twitter, Github, etc... It did not include integration with Identity Server as we cannot include a dependency on a dead-end product, nor can we use its replacement because the license is incompatiible with Oqtane.

**Reply by markdav-is** [reply]
I'm interested in the comment about ID server being a dead-end.  I need to do some research as we are still using it for new projects.

**Comment by curlyfro** [comment]
yikes!  -- ok

i'm heavily invested in Identity Server.  i took a look at the Oqtane auth code and I know it's a heavy lift.  I hope the "Authentication extensibility" abstraction will allow me to create a custom module and connect to my IDS.  so you're thinking a Q1 release?

**Reply by acpt** [reply]
If this happens - "Oqtane auth with Autentication exensibility", will make Oqtane framework an interesting replacement for Identity Server with many good extras
